clic.adoc

```diff
@@ -387,9 +387,45 @@ interrupt input.
 
 == smclic M-mode CLIC extension
 
-=== CLIC Memory-Mapped Registers
+Implementations can provide access to CLIC registers clicinttrig[i], clicintip[i], clicintie[i], clicintattr[i], clicintctl[i] 
+by memory-mapped registers or by using the Indirect CSR Access method (Smcsrind/Sscsrind).  
+The clic register accesses method available to software is indicated by the parameters CLICMEMMAPPED and CLICCSRINDIRECT.
+
+=== CLIC Register Access via Indirect CSR Access 
+
+The Indirect CSR Access allocated range for CLIC is TBD.  These address provided below are relative to the base of that allocated range.
 
-==== CLIC Memory Map
+[%autofit]
+----
+ miselect  |     mireg             |
+ offset    |register state         |
+
+  0x0       clicintattr[(XLEN/8)-1:0]
+  0x1       clicintattr[2*(XLEN/8)-1:(XLEN/8)]  
+  ...
+  0x400     clicintattr[(XLEN/8)-1:0]
+  0x101     clicintattr[2*(XLEN/8)-1:(XLEN/8)]
+  ...
+  0x800     clicintip[XLEN-1:0]       
+  0x801     clicintip[2*XLEN-1:XLEN]  
+  ...
+  0x880     clicintie[XLEN-1:0]       
+  0x881     clicintie[2*XLEN-1:XLEN]  
+  ...
+  0x900     clicinttrig[0]
+  0x901     clicinttrig[1]
+  ...
+  0x91F     clicinttrig[31}
+  0x920     reserved for smclicconfig extension
+----
```

Comment:
It seems that registers are only ever mapped to mireg, and mireg2-6 are unused. For configuration, that would cause quite some overhead. My suggestion would be to extend the table with five additional columns for mireg2-6, and to map registers there.

Reply:
We need to keep the indexing straightforward for software.  The only resource being consumed is miselect address space, and there is not that much being used, since there are 32 interrupts at each index.